libmamba/include/mamba/filesystem/u8path.hpp

```diff
@@ -1151,18 +1138,7 @@ namespace fs
     }
 
     // void last_write_time(const path& p, now _, error_code& ec) noexcept;
-    inline void last_write_time(const u8path& path, now, std::error_code& ec) noexcept
-    {
-#if defined(USE_UTIMENSAT)
-        if (utimensat(AT_FDCWD, path.string().c_str(), NULL, 0) == -1)
-        {
-            ec = std::error_code(errno, std::generic_category());
-        }
-#else
-        auto new_time = fs::file_time_type::clock::now();
-        std::filesystem::last_write_time(path, new_time, ec);
-#endif
-    }
+    void last_write_time(const u8path& path, now, std::error_code& ec) noexcept;
```

Comment:
Note that we used inlined functions specifically so that `<filesystem>` was used directly seemlessly at compilation for the user. While this specific function is not matching that intent (it was added afterwards), keep in mind that moving the code of functions in general from this file into cpp will have some (probably negative) effects on performance. As you didnt move other functions I suppose you realize all that already, just stating this "for the record".

Reply:
Yes, I move this one in particular because it had OS specific dependencies that I though were best hidden.